src/s3fs.cpp

```diff
@@ -3368,10 +3368,16 @@ static int readdir_multi_head(const char* path, const S3ObjList& head, void* buf
 
         for(s3obj_list_t::iterator reiter = notfound_param.notfound_list.begin(); reiter != notfound_param.notfound_list.end(); ++reiter){
             int dir_result;
-            if(0 == (dir_result = directory_empty(reiter->c_str()))){
+            std::string reiter_str = *reiter;
+            if('/' != *reiter_str.begin()){
+                // path must start with '/'
+                reiter_str = "/" + reiter_str;
+            }
+            dir_result = directory_empty(reiter_str.c_str());
+            if(0 == dir_result || -ENOTEMPTY == dir_result){
```

Comment:
Just to make sure I understand this: the previous implementation was effectively "if the directory is empty" but with this fix it's "is a directory" -- regardless if it's empty, just is deterministically directory-like. Correct?

Reply:
i think thats a good way to think about it